### PR TITLE
Use a variable for the username on coreos to fix the gce quickstart guide

### PR DESCRIPTION
**Pull request by roberthbailey**

### CODE REVIEW

**Comment by pbx0** [comment]
@roberthbailey Looks like the changes to add the cloud-provider support got clobbered in this commit maybe through rebase issues.

Otherwise this seems fine. Alternatively maybe we should update the gce-quickstart scripts and just ensure users add a public-key to their metadata that specifies the core user.

**Comment by roberthbailey** [comment]
> Alternatively maybe we should update the gce-quickstart scripts and just ensure users add a public-key to their metadata that specifies the core user.

That works for me. My main goal was that I wanted the quickstart to work when following the directions without needing to jump through extra hoops (e.g. debugging and editing bash). 

It's up to you as to which way you want to fix it. Let me know if you want me to fix up this PR or if you'd rather fix the quickstart guide for how to properly add an ssh key to the created VMs so that it's possible to ssh in as the `core` user.

**Comment by pbx0** [comment]
@roberthbailey Lets go with this PR.  I'll merge once fixed up.

**Comment by roberthbailey** [comment]
@pbx0 - which changes were clobbered? The diff looks right to me (essentially replacing core with a variable everywhere and defaulting the variable to core).

File: hack/quickstart/init-master.sh [hunk 1]
```diff
@@ -56,29 +56,26 @@ function init_master_node() {
...
+    cp /home/${REMOTE_USER}/assets/auth/kubeconfig /etc/kubernetes/
+    cp /home/${REMOTE_USER}/assets/tls/ca.crt /etc/kubernetes/ca.crt
 
     # Start etcd.
     if [ "$SELF_HOST_ETCD" = false ] ; then
         configure_etcd
         systemctl enable etcd-member; sudo systemctl start etcd-member
     fi
 
-    # Set cloud provider
```
Comment (pbx0) [inline hack/quickstart/init-master.sh]: It looks like some lines here got removed unrelated to this PR. I think basically a few changes from https://github.com/kubernetes-incubator/bootkube/commit/bc9154ea8c403425f9a69e0ea8385f1bda40e3f1 got reverted. It added cloud provider support to the quickstart scripts.

File: hack/quickstart/init-master.sh [hunk 2]
```diff
@@ -3,11 +3,11 @@ set -euo pipefail
 
 REMOTE_HOST=$1
 REMOTE_PORT=${REMOTE_PORT:-22}
+REMOTE_USER=${REMOTE_USER:-core}
 CLUSTER_DIR=${CLUSTER_DIR:-cluster}
 IDENT=${IDENT:-${HOME}/.ssh/id_rsa}
 SSH_OPTS=${SSH_OPTS:-}
 SELF_HOST_ETCD=${SELF_HOST_ETCD:-false}
-CLOUD_PROVIDER=${CLOUD_PROVIDER:-}
```
Comment (pbx0) [inline hack/quickstart/init-master.sh]: Cloud provider var removed here

File: hack/quickstart/init-master.sh [hunk 3]
```diff
@@ -92,22 +89,22 @@ function init_master_node() {
...
+    scp -i ${IDENT} -P ${REMOTE_PORT} ${SSH_OPTS} kubelet.master ${REMOTE_USER}@${REMOTE_HOST}:/home/${REMOTE_USER}/kubelet.master
+    ssh -i ${IDENT} -p ${REMOTE_PORT} ${SSH_OPTS} ${REMOTE_USER}@${REMOTE_HOST} "sudo mv /home/${REMOTE_USER}/kubelet.master /etc/systemd/system/kubelet.service"
 
     # Copy bootkube binary to remote host.
-    scp -i ${IDENT} -P ${REMOTE_PORT} ${SSH_OPTS} ../../_output/bin/linux/bootkube core@${REMOTE_HOST}:/home/core/bootkube
+    scp -i ${IDENT} -P ${REMOTE_PORT} ${SSH_OPTS} ../../_output/bin/linux/bootkube ${REMOTE_USER}@${REMOTE_HOST}:/home/${REMOTE_USER}/bootkube
 
     # Copy self to remote host so script can be executed in "local" mode
-    scp -i ${IDENT} -P ${REMOTE_PORT} ${SSH_OPTS} ${BASH_SOURCE[0]} core@${REMOTE_HOST}:/home/core/init-master.sh
-    ssh -i ${IDENT} -p ${REMOTE_PORT} ${SSH_OPTS} core@${REMOTE_HOST} "sudo CLOUD_PROVIDER=${CLOUD_PROVIDER} SELF_HOST_ETCD=${SELF_HOST_ETCD} /home/core/init-master.sh local"
```
Comment (pbx0) [inline hack/quickstart/init-master.sh]: same here

**Comment by roberthbailey** [comment]
@pbx0 - good eye; I'd missed that those lines were clobbered. I've replaced the CLOUD_PROVIDER bits in both files and taken a closer look at the diff this time. 

PTAL.

**Comment by pbx0** [comment]
rktbot run e2e

**Comment by pbx0** [comment]
rktbot run e2e

**Pull request merged**